src/renderer/js/redux/actions/shape_shift.js

```diff
@@ -0,0 +1,137 @@
+// @flow
+import Promise from "bluebird";
+import * as types from "constants/action_types";
+import { coinRegexPatterns } from "util/shape_shift";
+import type {
+  GetSupportedCoinsSuccess,
+  GetCoinStatsStart,
+  GetCoinStatsSuccess,
+  GetCoinStatsFail,
+  PrepareShapeShiftSuccess,
+  PrepareShapeShiftFail,
+  GetActiveShiftSuccess,
+  GetActiveShiftFail,
+} from "redux/reducers/shape_shift";
+import type { FormikActions } from "types/common";
+
+// use promise chains instead of callbacks for shapeshift api
+const shapeShift = Promise.promisifyAll(require("shapeshift.io"));
+
+// All ShapeShift actions
+// Action types defined in the reducer will contain some payload
+export type Action =
+  | { type: types.GET_SUPPORTED_COINS_START }
+  | { type: types.GET_SUPPORTED_COINS_FAIL }
+  | GetSupportedCoinsSuccess
+  | GetCoinStatsStart
+  | { type: types.GET_COIN_STATS_START }
+  | GetCoinStatsFail
+  | GetCoinStatsSuccess
+  | { type: types.PREPARE_SHAPE_SHIFT_START }
+  | PrepareShapeShiftFail
+  | PrepareShapeShiftSuccess
+  | { type: types.GET_ACTIVE_SHIFT_START }
+  | GetActiveShiftFail
+  | GetActiveShiftSuccess;
+
+// Basic thunk types
+// It would be nice to import these from types/common
+// Not sure how that would work since they rely on the Action type
+type PromiseAction = Promise<Action>;
```

Comment:
can you make these generic and move them to a different folder as you alluded to in the comment? I also am not a huge fan of using the any type here but for now I think this is good.

i.e.

```type PromiseAction<T> = Promise<T>;
 type ThunkAction<T> = (dispatch: Dispatch<T>) => any;
 export type Dispatch<T> = (
   action: T | ThunkAction<T> | PromiseAction<T> | Array<T>
 ) => any;

Reply:
I don't want you to get bogged down with this if for some reason its really difficult